worlds/mmbn3/__init__.py

```diff
@@ -74,10 +79,11 @@ def generate_early(self) -> None:
         if self.options.extra_ranks > 0:
             self.item_frequencies[ItemName.Progressive_Undernet_Rank] = 8 + self.options.extra_ranks
 
+        self.excluded_locations = []
+        if not self.options.include_secret:
+            self.excluded_locations = secret_locations
```

Comment:
This could use `+=` in case you ever change this around for ordering or anything else
```suggestion
            self.excluded_locations += secret_locations
```